homeassistant/components/system_bridge/sensor.py

```diff
@@ -51,88 +54,216 @@ class SystemBridgeSensorEntityDescription(SensorEntityDescription):
+    if (per_cpu := data.cpu.per_cpu) is not None and (index < len(per_cpu)):
+        return per_cpu[index].power
     return None
 
 
-def cpu_power_per_cpu(
+def cpu_usage_per_cpu(
     data: SystemBridgeCoordinatorData,
-    cpu: int,
+    index: int,
 ) -> float | None:
-    """Return CPU power per CPU."""
-    if (value := getattr(data.cpu, f"power_per_cpu_{cpu}", None)) is not None:
-        return value
+    """Return CPU usage per CPU."""
+    if (per_cpu := data.cpu.per_cpu) is not None and (index < len(per_cpu)):
+        return per_cpu[index].usage
     return None
 
 
 def cpu_speed(data: SystemBridgeCoordinatorData) -> float | None:
     """Return the CPU speed."""
-    if data.cpu.frequency_current is not None:
-        return round(data.cpu.frequency_current / 1000, 2)
+    if (cpu_frequency := data.cpu.frequency) is not None and (
+        cpu_frequency.current
+    ) is not None:
+        return round(cpu_frequency.current / 1000, 2)
+    return None
+
+
+def get_display(
+    data: SystemBridgeCoordinatorData,
+    index: int,
+) -> Display | None:
+    """Return the Display."""
+    if index < len(data.displays):
+        return data.displays[index]
+    return None
+
+
+def display_resolution_horizontal(
+    data: SystemBridgeCoordinatorData,
+    index: int,
+) -> int | None:
+    """Return the Display resolution horizontal."""
+    if (display := get_display(data, index)) is not None:
+        return display.resolution_horizontal
     return None
 
 
-def gpu_core_clock_speed(data: SystemBridgeCoordinatorData, key: str) -> float | None:
+def display_resolution_vertical(
+    data: SystemBridgeCoordinatorData,
+    index: int,
+) -> int | None:
+    """Return the Display resolution vertical."""
+    if (display := get_display(data, index)) is not None:
+        return display.resolution_vertical
+    return None
+
+
+def display_refresh_rate(
+    data: SystemBridgeCoordinatorData,
+    index: int,
+) -> float | None:
+    """Return the Display refresh rate."""
+    if (display := get_display(data, index)) is not None:
+        return display.refresh_rate
+    return None
+
+
+def get_gpu(
+    data: SystemBridgeCoordinatorData,
+    index: int,
+) -> GPU | None:
+    """Return the GPU."""
+    if index < len(data.gpus):
+        return data.gpus[index]
+    return None
+
+
+def gpu_core_clock_speed(
+    data: SystemBridgeCoordinatorData,
+    index: int,
+) -> float | None:
     """Return the GPU core clock speed."""
-    if (value := getattr(data.gpu, f"{key}_core_clock", None)) is not None:
-        return round(value)
+    if (gpu := get_gpu(data, index)) is not None:
+        return gpu.core_clock
     return None
 
 
-def gpu_memory_clock_speed(data: SystemBridgeCoordinatorData, key: str) -> float | None:
+def gpu_fan_speed(
+    data: SystemBridgeCoordinatorData,
+    index: int,
+) -> float | None:
+    """Return the GPU fan speed."""
+    if (gpu := get_gpu(data, index)) is not None:
```

Comment:
This exact check is duplicated 16 times. Maybe make that check in an outer function or decorator? Then this function would be something like:

@needs_gpu
def gpu_fan_speed(
    data: SystemBridgeCoordinatorData,
    index: int,
) -> float | None:
    """Return the GPU fan speed."""
    return gpu.fan_speed